frontend/src/components/pages/FridgeManagement/FridgeCheckIns/EditCheckin.tsx

```diff
@@ -0,0 +1,164 @@
+import {
+  Button,
+  Center,
+  CloseButton,
+  Container,
+  Flex,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  HStack,
+  Input,
+  Spinner,
+  Text,
+  Textarea,
+  useToast,
+} from "@chakra-ui/react";
+import { format, isAfter, isEqual, parse } from "date-fns";
+import React, { useEffect, useState } from "react";
+import { useHistory, useParams } from "react-router-dom";
+
+import CheckInAPIClient from "../../../../APIClients/CheckInAPIClient";
+import * as Routes from "../../../../constants/Routes";
+import { UpdateCheckInFields } from "../../../../types/CheckInTypes";
+import ErrorMessages from "./ErrorMessages";
+
+const EditCheckInPage = (): JSX.Element => {
+  const { id } = useParams<{ id: string }>();
+  const [date, setDate] = useState<string>("");
+  const [startTime, setStartTime] = useState<string>("");
+  const [endTime, setEndTime] = useState<string>("");
+  const [notes, setNotes] = useState<string | undefined>("");
+  const [timeRangeError, setTimeRangeError] = useState("");
+  const toast = useToast();
+  const history = useHistory();
+
+  const getCheckInData = async () => {
+    const checkInResponse = await CheckInAPIClient.getCheckInsById(id);
+    setDate(checkInResponse.startDate);
+    setStartTime(format(new Date(checkInResponse.startDate), "kk:mm"));
+    setEndTime(format(new Date(checkInResponse.endDate), "kk:mm"));
+    setNotes(checkInResponse.notes);
+  };
+
+  const validateForm = () => {
+    let newError = "";
+    let valid = true;
+    const parsedStartTime = parse(startTime, "kk:mm", new Date());
+    const parsedEndTime = parse(endTime, "kk:mm", new Date());
+
+    if (isAfter(parsedStartTime, parsedEndTime)) {
+      valid = false;
+      newError = ErrorMessages.endTimeBeforeStartTime;
+    } else if (isEqual(parsedStartTime, parsedEndTime)) {
+      valid = false;
+      newError = ErrorMessages.endTimeEqualsStartTime;
+    }
+
+    setTimeRangeError(newError);
+    return valid;
+  };
+
+  const onSaveClick = async () => {
+    if (!validateForm()) {
+      return;
+    }
+
+    const checkInData: UpdateCheckInFields = {
+      startDate: parse(startTime, "kk:mm", new Date(date)).toString(),
+      endDate: parse(endTime, "kk:mm", new Date(date)).toString(),
+      notes,
+    };
+
+    const res = await CheckInAPIClient.updateCheckInById(id, checkInData);
+
+    if (!res) {
+      toast({
+        title: "Checkin could not be updated. Please try again",
+        status: "error",
+        duration: 7000,
+        isClosable: true,
+      });
+      return;
+    }
+    toast({
+      title: "Check-in has been successfully updated",
+      status: "success",
+      duration: 7000,
+      isClosable: true,
+    });
+    history.push(Routes.ADMIN_CHECK_INS);
+  };
+
+  useEffect(() => {
+    getCheckInData();
+  }, [id]);
```

Comment:
really small nit, but this id is only changed if the component is reloaded so no need to have it as a dependency since the side effect would run anyways.

```suggestion
  }, []);
```